sql: remove unused dropOwnedByNode

mgartner opened the pull request:
`DROP OWNED BY` is supported only in the declarative schema changer and
`dropOwnedByNode` is not used, so it has been removed.

Epic: None

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/137739)
<!-- Reviewable:end -->

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/137739#-:-OESF79u3C5uW3LrPSV9:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @mgartner)



<!-- Sent from Reviewable.io -->

rafiss reviewed (changes requested):
i'm not sure if we should go with this PR. it would cause the following noisy behavior in explicit transactions:
```
root@localhost:26257/defaultdb> begin;
BEGIN

root@localhost:26257/defaultdb  OPEN> drop owned by root;
ERROR: internal error: unknown opaque statement *tree.DropOwnedBy
SQLSTATE: XX000
DETAIL: stack trace:
pkg/sql/opaque.go:312: planOpaque()
pkg/sql/opaque.go:59: buildOpaque()
pkg/sql/opt/optbuilder/opaque.go:65: tryBuildOpaque()
pkg/sql/opt/optbuilder/builder.go:483: buildStmt()
pkg/sql/opt/optbuilder/builder.go:316: buildStmtAtRootWithScope()
pkg/sql/opt/optbuilder/builder.go:297: buildStmtAtRoot()
pkg/sql/opt/optbuilder/builder.go:276: Build()
pkg/sql/plan_opt.go:824: buildExecMemo()
pkg/sql/plan_opt.go:260: makeOptimizerPlan()
pkg/sql/conn_executor_exec.go:3187: makeExecPlan()
pkg/sql/conn_executor_exec.go:2729: dispatchToExecutionEngine()
pkg/sql/conn_executor_exec.go:1025: execStmtInOpenState()
pkg/sql/conn_executor_exec.go:164: func2()
pkg/sql/conn_executor_exec.go:4334: execWithProfiling()
pkg/sql/conn_executor_exec.go:163: execStmt()
pkg/sql/conn_executor.go:2381: func1()
pkg/sql/conn_executor.go:2386: execCmd()
pkg/sql/conn_executor.go:2303: run()
pkg/sql/conn_executor.go:999: ServeConn()
pkg/sql/pgwire/conn.go:252: processCommands()
pkg/sql/pgwire/server.go:1197: func4()
src/runtime/asm_arm64.s:1222: goexit()
```

a related issue is https://github.com/cockroachdb/cockroach/issues/137615; we want to move all these statements that are unimplemented in the legacy schema changer to one place.

mgartner committed:
sql: remove unused dropOwnedByNode

`DROP OWNED BY` is supported only in the declarative schema changer and
`dropOwnedByNode` is not used, so it has been removed.

Release note: None

mgartner commented:
Ahh, good catch @rafiss. I changed the PR to just remove the plan node, not the unimplemented error code path.

rafiss reviewed (approved):
thanks for the cleanup!

> Reviewed 1 of 4 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/137739#-:-OEUPsAb8Gr2H3GzUAeA:bj9u4r9)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @mgartner)



<!-- Sent from Reviewable.io -->

mgartner commented:
TFTRs!

bors r+
